File: ios/GPS/bridge/SecureStorageManager.m
//
//  SecureStorageManager.m
//  COVIDSafePaths
//
//  Created by Tyler Roach on 4/29/20.
//  Copyright © 2020 Path Check Inc. All rights reserved.
//

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(SecureStorageManager, NSObject)

RCT_EXTERN_METHOD(
                  getLocations: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
)

RCT_EXTERN_METHOD(
                  importGoogleLocations: (NSArray)locations
                  resolve: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
)

RCT_EXTERN_METHOD(
                  migrateExistingLocations: (NSArray)locations
                  resolve: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
)
@end

File: ios/GPS/bridge/SecureStorageManager.swift
//
//  SecureStorageManager.swift
//  COVIDSafePaths
//
//  Created by Tyler Roach on 4/29/20.
//  Copyright © 2020 Path Check Inc. All rights reserved.
//

import Foundation

@objc(SecureStorageManager)
class SecureStorageManager: NSObject {
  
  @objc static func requiresMainQueueSetup() -> Bool {
      return false
  }
  
  @objc
  func getLocations(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    GPSSecureStorage.shared.getLocations(resolve: resolve, reject: reject)
  }
  
  @objc
  func importGoogleLocations(_ locations: NSArray, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    GPSSecureStorage.shared.importLocations(locations: locations, source: .google, resolve: resolve, reject: reject)
  }
  
  @objc
  func migrateExistingLocations(_ locations: NSArray, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    GPSSecureStorage.shared.importLocations(locations: locations, source: .migration, resolve: resolve, reject: reject)
  }

  @objc
  func trimLocations(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    GPSSecureStorage.shared.getLocations(resolve: resolve, reject: reject)
  }

}
